web-client/src/HabitUtil.elm

```diff
@@ -0,0 +1,188 @@
+module HabitUtil exposing (..)
+
+{-| Module for useful Habit operations
+-}
+
+import Models.Habit as Habit
+import Models.FrequencyStats as FrequencyStats
+
+
+isHabitCurrentFragmentSuccessful : Habit.Habit -> FrequencyStats.FrequencyStats -> Bool
+isHabitCurrentFragmentSuccessful habit frequencyStats =
+    case habit of
+        Habit.GoodHabit _ ->
+            frequencyStats.currentFragmentTotal >= frequencyStats.currentFragmentGoal
+
+        Habit.BadHabit _ ->
+            frequencyStats.currentFragmentTotal <= frequencyStats.currentFragmentGoal
+
+
+findFrequencyStatsForHabit : Habit.Habit -> List FrequencyStats.FrequencyStats -> Result String FrequencyStats.FrequencyStats
+findFrequencyStatsForHabit habit frequencyStats =
```

Comment:
This function should probably just be one line, just use forward pipes:

```elm
List.filter (\stats -> stats.habitId == habitId) frequencyStats
    |> List.head
    ...
``` 

You don't need to check here for duplicate frequency stats, just take 1, I think we're safe to not worry about sending copies